### PR TITLE
Parse and compile with extra build flags

### PR DESCRIPTION
**Pull request by nc-holodakg**

### CODE REVIEW

File: internal/lambgofile/lambgofile.go [hunk 1]
```diff
@@ -62,6 +68,8 @@ type Config struct {
 
 	OutDirectory   string   `yaml:"outDirectory"`
 	ZippedFileName string   `yaml:"zippedFileName"`
+	RawExtraFlags  string   `yaml:"extraFlags"`
```
Comment (JosiahWitt) [inline internal/lambgofile/lambgofile.go:71]: NP: Maybe call these `extraBuildFlags` or simply `buildFlags` for clarity, so it doesn't get confused with flags passed to Lambgo?

Comment (JosiahWitt) [inline internal/lambgofile/lambgofile.go:71]: Alternatively, if we don't want to add the `shlex` dependency, we could do the approach which lots of Go tools use, and just accept an array. I believe Docker does this. And something sorta closer to this project is GoReleaser. Here's their [config file](https://goreleaser.com/customization/build/), where they just have a `flags` array that is passed directly to `go build`.

File: go.sum [hunk 2]
```diff
@@ -32,7 +34,6 @@ golang.org/x/sync v0.0.0-20190423024810-112230192c58/go.mod h1:RxMgew5VJxzue5/jJ
 golang.org/x/sys v0.0.0-20190215142949-d0b11bdaac8a/go.mod h1:STP8DvDyc/dI5b8T5hshtkjS+E42TnysNCUPdjciGhY=
 golang.org/x/sys v0.0.0-20190412213103-97732733099d/go.mod h1:h1NjWce9XRLGQEsW7wpKNCjG9DtNlClVuFLEZdDNbEs=
 golang.org/x/text v0.3.0/go.mod h1:NqM8EUOU14njkJ3fqMW+pc6Ldnwhi/IjpwHt7yyuwOQ=
-golang.org/x/tools v0.0.0-20191119224855-298f0cb1881e h1:aZzprAO9/8oim3qStq3wc1Xuxx4QmAGriC4VU4ojemQ=
```
Comment (nc-holodakg) [inline go.sum:35]: This is from `go mod tidy`, so I assume it should be committed?

File: internal/lambgofile/lambgofile_test.go [hunk 3]
```diff
@@ -105,6 +105,26 @@ zippedFileName: some-name
 			}),
 		},
 
+		{
+			Name: "with valid config include buildFlags",
```
Comment (JosiahWitt) [inline internal/lambgofile/lambgofile_test.go:109]:
```suggestion
			Name: "with valid config including buildFlags",
```

File: internal/lambgofile/lambgofile_test.go [hunk 4]
```diff
@@ -168,6 +188,17 @@ zippedFileName: some-name
 				"my/app/.lambgo.yml": "{{{{{{ Not YAML",
 			}),
 		},
+
+		{
+			Name:          "when .lambgo.yml file has bad flag synxtax",
```
Comment (JosiahWitt) [inline internal/lambgofile/lambgofile_test.go:193]:
```suggestion
			Name:          "when .lambgo.yml file has bad flag syntax",
```

**Review by JosiahWitt — APPROVED** [review]

File: go.sum (hunk 2 above)
Comment (JosiahWitt) [reply go.sum:35]: Yep!

**Pull request merged**